### PR TITLE
feat: Adds a graph parameter to specify the named graph used for shapes.

### PR DESCRIPTION
**Pull request by cristianvasquez**
Implements:

https://github.com/zazuko/barnard59-rdf/issues/28

### CODE REVIEW

File: lib/cube/buildCubeShape/index.js [hunk 1]
```diff
@@ -92,15 +93,20 @@ class CubeShapeBuilder extends Transform {
 
   _flush (callback) {
     for (const cube of this.options.cubes.values()) {
-      this.push(cube.toDataset())
+      const dataset = cube.toDataset({ shapeGraph: toNamedNode(this.options.graph) })
+      this.push(dataset)
```
Comment (tpluscode) [inline lib/cube/buildCubeShape/index.js:97]: I think this PR is unnecessarily complicated. I would revert changes in the `Cube` class and simply map the whole result to the selected graph

```suggestion
      const dataset = cube.toDataset()
      const graph = toNamedNode(this.options.graph) || $rdf.defaultGraph()

      this.push(dataset.map({ subject, predicate, object }) => 
        $rdf.quad(subject, predicate, object, graph))
```

**Review by tpluscode — CHANGES_REQUESTED** [review]

File: lib/cube/buildCubeShape/index.js (hunk 1 above)
Comment (cristianvasquez) [reply lib/cube/buildCubeShape/index.js:97]: That was the initial implementation, 

The only problem is some triples belong to something other than a shape, and will end up with the shape the named-graph. 

For example:

<obs> a cube:ObservationSet .

Comment (tpluscode) [reply lib/cube/buildCubeShape/index.js:97]: ah ok. the confusing part is that the module is called `buildCubeShape` although it does more than just cube. let me have another look

File: lib/cube/buildCubeShape/Cube.js [hunk 2]
```diff
@@ -49,15 +49,18 @@ class Cube {
     clownface({ dataset, term: this.observationSet })
       .addOut(ns.rdf.type, ns.cube.ObservationSet)
 
-    clownface({ dataset, term: this.shape })
+    const shapeDataset = rdf.dataset()
+
+    clownface({ dataset: shapeDataset, term: this.shape })
```
Comment (tpluscode) [inline lib/cube/buildCubeShape/Cube.js:54]: I think the only real improvement could be to get rid of the `setGraph` function and `shapeDataset` if you initialise the pointer with a named graph

```suggestion
    clownface({ dataset, term: this.shape, graph: shapeGraph })
```

File: lib/cube/buildCubeShape/Cube.js [hunk 3]
```diff
@@ -36,7 +36,7 @@ class Cube {
     this.dimension({ predicate, object }).update({ predicate, object })
   }
 
-  toDataset () {
+  toDataset ({ shapeGraph } = { shapeGraph: undefined }) {
```
Comment (tpluscode) [inline lib/cube/buildCubeShape/Cube.js:39]:
```suggestion
  toDataset ({ shapeGraph } = { }) {
```

`undefined` is unnecessary

**Review by tpluscode — CHANGES_REQUESTED** [review]

File: lib/cube/buildCubeShape/Cube.js (hunk 2 above)
Comment (cristianvasquez) [reply lib/cube/buildCubeShape/Cube.js:54]: But later one adds datasets produced by the dimensions, 

```js
    for (const dimension of this.dimensions.values()) {
      shapeDataset.addAll(dimension.toDataset({ shape: this.shape }))
    }
    ```
    
    Apparently that part does not respect the graphName used in the clownface object (shapeDataset)

Comment (tpluscode) [reply lib/cube/buildCubeShape/Cube.js:54]: Uh, ok, I have no more comments then. Let's keep it as is it

**Pull request merged**